### PR TITLE
Added some new tests

### PR DESCRIPTION
**Pull request by tommorris**
Just [being bold](https://en.wikipedia.org/wiki/Wikipedia:BOLD) and adding some new tests and docstrings.

### CODE REVIEW

File: tests/app/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+from app.utils import (display_list,
+                       strip_whitespace_from_data,
+                       json_has_matching_id,
+                       json_has_required_keys,
+                       link)
+
+from nose.tools import assert_equal, raises
```
Comment (allait) [inline tests/app/test_utils.py]: API was switched to pytest, so we should probably start using pytest tools for new tests

File: tests/app/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+def test_strip_whitespace_from_data():
+    struct = {"eggs": "  ham "}
+    assert_equal(strip_whitespace_from_data(struct)['eggs'], "ham")
+    struct_with_list = {"eggs": ["  spam  ", "  ham  ", "  eggs "]}
+    after_strip = strip_whitespace_from_data(struct_with_list)
+    for item in after_strip['eggs']:
+        # check to see if there is any whitespace left.
+        # if not, find will return -1
+        assert_equal(item.find(" "), -1)
```
Comment (allait) [inline tests/app/test_utils.py]: `assert " " not in item`

**Comment by allait** [comment]
I don't think we've ever officially adopted it, but I think most of our commit messages follow the GDS styleguide (https://github.com/alphagov/styleguides/blob/master/git.md), so it might be worth taking a look at it.

File: app/utils.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 
 
 def link(rel, href):
+    """Generates a link dict from a rel, href pair."""
```
Comment (allait) [inline app/utils.py]: Should be "Generate a link ...", according to https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

File: tests/app/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+from app.utils import (display_list,
```
Comment (allait) [inline tests/app/test_utils.py]: Again, I don't think we've ever agreed on this, but things that are tested usually are imported after the test tools.

**Comment by robyoung** [comment]
Needs a rebase but :+1: from me.

**Pull request closed without merging**